### ISSUE TITLE
`prefer-range` combines adjacent but not-allowed ranges.

### ISSUE THREAD

**Issue by RunDevelopment** [ota-meshi/eslint-plugin-regexp#218]
With the settings:

```js
settings: {
	regexp: {
		// allow alphanumeric and cyrillic ranges
		allowedCharacterRanges: ['alphanumeric', 'а-я', 'А-Я']
	}
},
```

`prefer-range` will combine `[а-яА-Я]` into `[А-я]` which then gets reported by `no-obscure-range` as an error. Clearly, `perfer-range` is a little too eager.
